opensrp-reveal/src/test/java/org/smartregister/reveal/fragment/AvailableOfflineMapsFragmentTest.java

```diff
@@ -0,0 +1,238 @@
+package org.smartregister.reveal.fragment;
+
+import android.content.Context;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.CheckBox;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.powermock.reflect.Whitebox;
+import org.robolectric.Robolectric;
+import org.robolectric.RuntimeEnvironment;
+import org.smartregister.domain.Location;
+import org.smartregister.reveal.BaseUnitTest;
+import org.smartregister.reveal.R;
+import org.smartregister.reveal.adapter.AvailableOfflineMapAdapter;
+import org.smartregister.reveal.contract.OfflineMapDownloadCallback;
+import org.smartregister.reveal.model.OfflineMapModel;
+import org.smartregister.reveal.presenter.AvailableOfflineMapsPresenter;
+import org.smartregister.reveal.util.TestingUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.emory.mathcs.backport.java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.verify;
+import static org.smartregister.reveal.model.OfflineMapModel.OfflineMapStatus.DOWNLOADED;
+import static org.smartregister.reveal.model.OfflineMapModel.OfflineMapStatus.DOWNLOAD_STARTED;
+import static org.smartregister.reveal.model.OfflineMapModel.OfflineMapStatus.READY;
+
+/**
+ * Created by Richard Kareko on 1/27/20.
+ */
+
+public class AvailableOfflineMapsFragmentTest extends BaseUnitTest {
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    @Mock
+    private AvailableOfflineMapsPresenter presenter;
+
+    @Mock
+    private AvailableOfflineMapAdapter adapter;
+
+    @Mock
+    private OfflineMapDownloadCallback callback;
+
+    @Captor
+    private ArgumentCaptor<List<OfflineMapModel>> offlineMapModelListCaptor;
+
+    @Captor
+    private ArgumentCaptor<List<String>> stringListCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    private AvailableOfflineMapsFragment fragment;
+
+    private Context context = RuntimeEnvironment.application;
+
+    private AppCompatActivity activity;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'activity' could be replaced by a local variable.](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4217760)